imagecodecs/_jpeg8.pyx

```diff
@@ -141,7 +141,7 @@ def jpeg8_encode(
         (src.dtype == numpy.uint8 or src.dtype == numpy.uint16)
         and src.ndim in {2, 3}
         # src.nbytes <= 2147483647 and  # limit to 2 GB
-        and samples in {1, 3, 4}
+        and samples in {1, 2, 3, 4}
```

Comment:
Seems to work as expected with 2 components:
```python3
import sys

from imagecodecs import imread, jpeg8_decode, jpeg8_encode
from numpy.testing import assert_array_equal

filename = sys.argv[1]

image = imread(filename)
enc = jpeg8_encode(
    image,
    lossless=True,
    predictor=1,
    bitspersample=16,
)
dec = jpeg8_decode(enc)
assert_array_equal(image, dec)
```
Using this input file: ![16](https://github.com/cgohlke/imagecodecs/assets/8997425/eae5cdb1-896e-4ce9-9e28-02c856e05b1a)